benchmark/buffers/buffer-indexof.js

```diff
@@ -37,10 +43,6 @@ function main({ n, search, encoding, type }) {
     encoding = undefined;
   }
 
-  if (encoding === 'ucs2') {
```

Comment:
Why is this being removed?

Reply:
IIUC, it was add specially for `usc2` benchmark originally in #2539:

> Added benchmark shows the expected improvements
Added option to use ucs2 encoding with Buffer::IndexOf

Now I added more `encoding` options, so I don't think it should be treated specially anymore.

Reply:
This code is necessary for multi-byte encodings like ucs2/utf16 because the `aliceBuffer.indexOf(search, 0, encoding)` will always fail to match anything otherwise if the `aliceBuffer` is not formed correctly (e.g. two bytes per character).

If anything we should be keeping this code and adding in the other multi-byte encodings into the check, something like:
```js
if (/^(ucs-?2|utf-?16le)$/i.test(encoding)) {
  // ....
}
```